### PR TITLE
Implement Thumbnail component

### LINKED ISSUE #898: is there a plan for page thumbnail?
### Before you start - checklist

- [X] I understand that React-PDF does not aim to be a fully-fledged PDF viewer and is only a tool to make one
- [X] I have checked if this feature request is not already reported

### Description

I understand that React-PDF does not aim to be a fully-fledged PDF viewer and is only a tool to make one, but since mostly all PDF viewer have page thumbnails for preview, i think it will be helpful to provide Thumbnail component. The Thumbnail component's usage could just like the Page component(be placed inside <Document />, provide pageIndex or pageNumber to choose which page's thumbnail to render...).

### Proposed solution

I have read the source code about how to  [generate thumbnails](https://github.com/mozilla/pdf.js/blob/master/web/pdf_thumbnail_view.js) in pdf.js and some code of react-pdf, if you think my idea is adoptable, I really hope you can give me a chance to work on this:)

### Alternatives

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by wojtekmaj**
Closes #898

### CODE REVIEW

**Comment by wojtekmaj** [comment]
TODO: Docs.

**Comment by wojtekmaj** [comment]
@iamandrewluca Let me know what you think!

**Review by iamandrewluca — COMMENTED** [review]
looks good to me 🙂

**Comment by wojtekmaj** [comment]
One thing I'm uneasy about is onClick prop, which is hardcoded to use linkService just like Outline links. Unlike Outline links, however, it does not provide accessibility features of a link.

File: src/Page/PageCanvas.tsx [hunk 1]
```diff
@@ -161,7 +162,7 @@ export default function PageCanvas(props: PageCanvasProps) {
 
   return (
     <canvas
-      className="react-pdf__Page__canvas"
+      className={`${_className}__canvas`}
```
[the remark this answers is not in the document]
Comment (wojtekmaj) [reply src/Page/PageCanvas.tsx:165]: Doesn't have to - it's always rendered by Page and is not exposed.

[the remark this answers is not in the document]
Comment (iamandrewluca) [reply src/Page/PageCanvas.tsx:165]: `PageCanvas` inside `Page` is rendered like this

```
return <PageCanvas key={`${pageKey}_canvas`} canvasRef={canvasRef} />;
```

No `_className` is passed to `PageCanvas`.

In the commented code we replaced `className="react-pdf__Page__canvas"` with `className={`${_className}__canvas`}` which means that now `_className` is supposed to be `react-pdf__Page`

[the remark this answers is not in the document]
Comment (iamandrewluca) [reply src/Page/PageCanvas.tsx:165]: Or I'm missing something 😄

[the remark this answers is not in the document]
Comment (wojtekmaj) [reply src/Page/PageCanvas.tsx:165]: You are - _className is passed to Context :D

**Pull request merged**